### PR TITLE
Fix inference with iohub

### PR DESCRIPTION
**Pull request by ziw-liu**
This branch consists of minor fixes that enable the inference in #209, but does not aim to be stable or performant.

With 505bba8941caadccb086a015e6a094c5fee33c7f I was able to confirm that nuclei prediction works with the A549 torch-gunpowder model.

### CODE REVIEW

File: micro_dl/torch_unet/networks/Unet2D.py [hunk 1]
```diff
@@ -207,8 +212,39 @@ def forward(self, x, validate_input=False):
         # output channel collapsing layer
         x = self.terminal_block(x)
 
+        #crop padded inputs to valid regions
+        if pad_nonmultiple_input:
+            x = x[...,0:-pad_shape[1], 0:-pad_shape[3]]
+            
         return x
 
+    def _pad_nonmultiple_input(self, x):
```
Comment (ziw-liu) [inline micro_dl/torch_unet/networks/Unet2D.py:221]: This method is exactly the same in 2D and 2.5D models (`diff -u`):
```diff
@@ -227,6 +271,7 @@
         :return torch.tensor x_padded: zero-padded x
         :return tuple pad_shape: shape x was padded by
         """
+        
         # zero-pad non multiple of 2**num_blocks shapes
         assert self.training == False, (
                 "Padding and cropping to shape requires evaluation mode.. "
```
We should consider a base or mixin class for these.

Comment (Christianfoley) [reply micro_dl/torch_unet/networks/Unet2D.py:221]: Yep. Been considering this from the start. Have avoided so far since there would only be two instances of the base-class type.

Comment (ziw-liu) [reply micro_dl/torch_unet/networks/Unet2D.py:221]: We can also make this a utility function. There is only one integer parameter (`num_blocks`) needed from `self`, and it does not modify the state of the object in any way. I can do this if you don't have conflicting local changes.

Comment (ziw-liu) [reply micro_dl/torch_unet/networks/Unet2D.py:221]: Actually I might take it further away into the inference module so that the switching logic is no longer in the model itself. See #214 for reasoning. @Christianfoley this will break your new tests and I'm open to any push back.

Comment (ziw-liu) [reply micro_dl/torch_unet/networks/Unet2D.py]: Changed in https://github.com/mehta-lab/microDL/pull/210/commits/7157d1f622309d5b83f2e2b9f63975a0d5ddef3c

**Review by Christianfoley — APPROVED** [review]
This all looks good to me. Thanks for moving the non-multiple handling to inference working through the all-sides padding logic.
